mbedtls/buildmbedtls

```diff
@@ -0,0 +1,18 @@
+#!/bin/sh
+set -ex
+
+MBEDTLS_SRC_ARCH=mbedtls-2.2.1
+
+# download mbedtls sources
+cd mbedtls
+wget https://github.com/ARMmbed/mbedtls/archive/$MBEDTLS_SRC_ARCH.tar.gz
+tar -xzf $MBEDTLS_SRC_ARCH.tar.gz
```

Comment:
I realize that this file mainly exists to set up testing on TravisCI, but this doesn't work on macOS because `wget` is not available by default. You could conditionally substitute `wget` with `curl -O -L`. Alternatively, you could replace this file with something in the top-level CMakeLists.txt with the `ExternalProject_Add` command. I'm winging this, but it would be something like:
```cmake
SET(MBEDTLS_SRC_ARCH mbedtls-2.2.1)
ExternalProject_Add(
  URL https://github.com/ARMmbed/mbedtls/archive/${MBEDTLS_SRC_ARC}.tar.gz
  PATCH patch -p0 < ../mbedtls.patch
  CMAKE_ARGS -DCMAKE_BUILD_TYPE=RelWithDebInfo -DUSE_SHARED_MBEDTLS_LIBRARY=ON
  BUILD_COMMAND make
  INSTALL_COMMAND make install
)
```